### ISSUE TITLE
Button on edit/create form

### ISSUE THREAD

**Issue by fabiograz** [r3-team/r3#10]
Hi,
when I click on new button (i.e. in Country list form in Location application) new form are displayed. In this form there is a "delete" button, but it is greyed. It become red only after saved the new record. I expected that it was an abort button, so if I don't want insert the record I can abort and return on the previous form (in this case the country list form). 
So also the "new" button is not useful in case I add a new record. In fact in this case if I insert a new record and press "new" I discard any inserted values.
I think that this buttons must better managed to simplify the user workflow.
For example the form button may be:
"new": in case there are any new values inserted by user a confirmation message must be showed so user may cancel operation
"save": as now, but if I modify a record a confirmation message must be displayed (this may be a configuration parameter) 
"Save + new": as now and as suggested for the "save" button
"Delete: as now, but after confirmation must be shown next/previews or last  record
"Exit": in this case, after a user confirmation in case of modified values on the form, application return to a predefined form (i.e. list)

In this manner the workflow is more clear. What do you think about?

Bye,
Fabio.

**Comment by r3-gabriel** [comment]
Hello Fabio,

With some suggestions I agree, others I think are more like personal preference. In order:
* In a form to create a new record
  * The `new` action is still available in case the user made bad inputs and wants to reset the form for a new record. I agree, that if inputs were made, a message should be shown, such as `Your inputs will be lost - are you sure?`.
  * The `delete` action is not necessary for new records but shown greyed out, because for some people it is preferable, when UI elements are at the same position in the UI, instead of appearing/changing depending on context (muscle memory and such).
* Changing the `delete` button to an `abort` button depending on the context (new or existing record) would mean that sometimes the "red button" will delete a record and sometimes it will return the user to a previous form. This button could be made to look differently but the error rate of misclicking might still be high. For this case, I would prefer to have a dedicated `go back` button on the form, which could give a warning if inputs were made.
* For the `save` buttons, I am a bit puzzled. Do you suggest a `Are you sure you want to save?` message on every save? That would be very annoying to me at least. Or do you mean a separate, unobtrusive message like `Your record was saved successfully` somewhere in the form header?

To summarize, I suggest:
* To keep the `new` button for new records to reset the inputs, but provide a warning message to the user, that existing inputs will be reset.
* To keep the `delete` button for new records, because it does not hurt and it is nice to have a consistent UI - but would add a dedicated `go back` button that would serve as `abort`/`exit` for new/existing records. With warning, if inputs would be lost.
* To implement a `Your record was saved successfully` message on save on the form. I am not sure, if you meant it this way, but it does make sense to have that message.
* The `exit` button´s functionality would be the same as the suggest `go back` button.

What do you think?

Warm regards,

Gabriel

**Comment by fabiograz** [comment]
Hi, 
I agree with your suggestions.  The message for the "save" button should be for informational purposes only and appear for a few seconds to indicate that the operation was successful (as well as in the event of an error).  For the exit button it is important that it can be defined which form to return to.
Best regard,
 Fabio

**Comment by r3-gabriel** [comment]
Hello Fabio,

then I will implement the changes for 2.5.1, which was already on its way to the finish line.

Regarding the exit/go-back button: The implementation I had in mind uses the browser history with REI3´s page router - which not only knows the previous form but would also set a list scroll position (if previous form was a list) to the last known value and recreate sorting and simple filters. Setting the form manually could be error prone - you can forget to change this and you can also reach the same form from 2 different places.

Warm regards,

Gabriel

**Comment by fabiograz** [comment]
Hello Gabriel, 
ok for your implementation. I've translated new messages to Italian

Thanks.
